tests/testcases/key-value-undefined-store/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::{ensure, Result};
+use itertools::sorted;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+    key_value::{Error, Store},
+};
+
+#[http_component]
+fn handle_request(req: Request) -> Result<Response> {
+    // TODO: once we allow users to pass non-default stores, test that opening
+    // an allowed-but-non-existent one returns Error::NoSuchStore
+    ensure!(matches!(Store::open("forbidden"), Err(Error::AccessDenied)));
+
+    let query = req.uri().query().expect("Should have a testkey query string");
+    let query: std::collections::HashMap::<String, String> = serde_qs::from_str(query)?;
+    let init_key = query.get("testkey").expect("Should have a testkey query string");
+    let init_val = query.get("testval").expect("Should have a testval query string");
+
+    let store = Store::open_default()?;
+
+    store.delete("bar")?;
+
+    ensure!(!store.exists("bar")?);
+
+    ensure!(matches!(store.get("bar"), Err(Error::NoSuchKey)));
+
+    store.set("bar", b"baz")?;
+
+    ensure!(store.exists("bar")?);
+
+    ensure!(b"baz" as &[_] == &store.get("bar")?);
```

Comment:
`assert_eq!` gives more useful errors, e.g.

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `4`,
 right: `3`', src/main.rs:5:1
```

I _think_ it would also remove the need to cast `as &[_]` in this case, but my confidence in understanding Rust's coercion rules is NOT HIGH.